### DISCUSSION TITLE
Running Artillery as a node library

### DISCUSSION THREAD

**Discussion by hsabillon7** [artilleryio/artillery#1043] [Artillery]
Hi just wanna know if there's any documentation on how to run Artillery inside a node project like a library or others modules. Because I only see docs about the CLI.

**Comment by hassy** [comment]
hi @hsabillon7 👋 it is possible to use Artillery as a library, but we don't recommend / support that type of usage officially.

**Reply by hsabillon7** [reply]
Thanks @hassy for the response! so that's why I didn't find any docs on how to do `artillery.run()` I wanna build one tool around artillery inside a NodeJS project but without official support/docs maybe I need to see if running the CLI process and using the JSON output will be a better bet

**Reply by hassy** [reply]
That's what we would typically recommend - treating Artillery as just another CLI. If you run into any issues, or have feedback on how that use-case could be improved, please share! :)
